### PR TITLE
docs/BUG-BOUNTY: proposed additional docs [ci skip]

### PR DESCRIPTION
**Pull request by bagder**
Bug bounty explainer. See https://bountygraph.com/programs/curl

This is documentation and an explainer with additional details from my bounty [proposal on the mailing list](https://curl.haxx.se/mail/lib-2018-09/0120.html) yesterday.

### CODE REVIEW

File: docs/BUG-BOUNTY.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ respective source codes - when running on existing hardware. It does not
+ include documentation, web sites or other infrastructure.
+
+ The curl security team will be the sole arbiter if a reported flaw can be
+ subject to a bounty or not.
+
+## How are vulnerabilities graded
+
+ The grading of each reported vulernability that makes a reward claim will be
+ performed by the curl security team. The grading should be based on the CVSS
```
Comment (richardthe3rd) [inline docs/BUG-BOUNTY.md]: "will be based on" instead of "should be"?

File: docs/BUG-BOUNTY.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ The curl security team first gives the vulnerability a score, as mentioned
+ above, and based on that level the team may increase or decrease the bounty
+ amount from the general template depending on the specifics of the individual
+ case.
+
+ The curl security team will be the sole arbiter of the bounty amount.
+
+## What happens if the bounty fund is drained
+
+ The boundy fund depends on sponsors. If we pay out more bounties than we add,
```
Comment (richardthe3rd) [inline docs/BUG-BOUNTY.md]: Typo: boundy->bounty

**Comment by bagder** [comment]
I don't think we can speak for other programs in which we have no say or influence. What do you think we should say about IBB? Mention that it exists (which we already do the in SECURITY-PROCESS doc) or state that we won't pay reward money for vulnerabilities already paid for by other bug bounties?

**Comment by bagder** [comment]
I don't think we should require "not getting paid by another bounty program" as a requirement. First, it makes it really hard to keep track of and secondly, a flaw is a flaw to us no matter if another program will pay for it or not and to the same extent. A reported security flaw that fulfills our requirement may be eligible for a bounty I think.

IBB has a requirement for paying a bounty that the reported flaw has to "be novel: vulnerability is new or unusual in an interesting way" - which we certyinly don't have as a requirement.

**Comment by jay** [comment]
Fair enough. This reminds me for readability I propose let's stop putting [ci skip] in the subject of commit messages and instead put it on its own line at the end of the body.

**Comment by bagder** [comment]
I was going to say that I thought one of those systems required the skip to be on that line. But now when I look it appears like both travis and appveyor still built this commit so it was a failure in several ways... :-/

File: docs/BUG-BOUNTY.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ the flaw is determined to be.
+
+ We offer reward money *up to* these amounts. The curl security team will
+ solely and exclusively determine the exact amount for each reported flaw on a
+ case by case basis and keep the rights to adjust the amount as it sees fit.
+
+ - Low         $500
+ - Medium    $1,000
+ - High      $5,000
+ - Critical $10,000
```
Comment (danielgustafsson) [inline docs/BUG-BOUNTY.md]: Nitpick, but should we use "USD 10,000" instead as $ is the denomination for many currencies (living in Australia taught me the confusions that can be had).

Comment (bagder) [reply docs/BUG-BOUNTY.md]: Makes sense yes, I'll amend.

**Pull request closed without merging**